cudax/include/cuda/experimental/__memory_resource/async_memory_resource.h

```diff
@@ -0,0 +1,298 @@
+  _CCCL_NODISCARD constexpr bool operator!=(async_memory_resource const& __rhs) const noexcept
+  {
+    return __pool_ != __rhs.__pool_;
+  }
+#    endif // _CCCL_STD_VER <= 2017
+
+  //! @brief Equality comparison between a async_memory_resource and another resource.
+  //! @param __lhs The async_memory_resource.
+  //! @param __rhs The resource to compare to.
+  //! @returns If the underlying types are equality comparable, returns the result of equality comparison of both
+  //! resources. Otherwise, returns false.
+  template <class _Resource>
+  _CCCL_NODISCARD_FRIEND auto operator==(async_memory_resource const& __lhs, _Resource const& __rhs) noexcept
+    _LIBCUDACXX_TRAILING_REQUIRES(bool)((_CUDA_VMR::__different_resource<async_memory_resource, _Resource>) )
+  {
+    return _CUDA_VMR::resource_ref<>{const_cast<async_memory_resource&>(__lhs)}
+        == _CUDA_VMR::resource_ref<>{const_cast<_Resource&>(__rhs)};
+  }
+#    if _CCCL_STD_VER <= 2017
+
+  //! @copydoc async_memory_resource::operator==<_Resource>(async_memory_resource const&, _Resource const&)
+  template <class _Resource>
+  _CCCL_NODISCARD_FRIEND auto operator==(_Resource const& __rhs, async_memory_resource const& __lhs) noexcept
+    _LIBCUDACXX_TRAILING_REQUIRES(bool)(_CUDA_VMR::__different_resource<async_memory_resource, _Resource>)
+  {
+    return _CUDA_VMR::resource_ref<>{const_cast<async_memory_resource&>(__lhs)}
+        == _CUDA_VMR::resource_ref<>{const_cast<_Resource&>(__rhs)};
+  }
+
+  //! @copydoc async_memory_resource::operator==<_Resource>(async_memory_resource const&, _Resource const&)
+  template <class _Resource>
+  _CCCL_NODISCARD_FRIEND auto operator!=(async_memory_resource const& __lhs, _Resource const& __rhs) noexcept
+    _LIBCUDACXX_TRAILING_REQUIRES(bool)(_CUDA_VMR::__different_resource<async_memory_resource, _Resource>)
+  {
+    return _CUDA_VMR::resource_ref<>{const_cast<async_memory_resource&>(__lhs)}
+        != _CUDA_VMR::resource_ref<>{const_cast<_Resource&>(__rhs)};
+  }
+
+  //! @copydoc async_memory_resource::operator==<_Resource>(async_memory_resource const&, _Resource const&)
+  template <class _Resource>
+  _CCCL_NODISCARD_FRIEND auto operator!=(_Resource const& __rhs, async_memory_resource const& __lhs) noexcept
+    _LIBCUDACXX_TRAILING_REQUIRES(bool)(_CUDA_VMR::__different_resource<async_memory_resource, _Resource>)
+  {
+    return _CUDA_VMR::resource_ref<>{const_cast<async_memory_resource&>(__lhs)}
+        != _CUDA_VMR::resource_ref<>{const_cast<_Resource&>(__rhs)};
+  }
+#    endif // _CCCL_STD_VER <= 2017
+
+  //! @brief Returns the underlying handle to the CUDA memory pool.
+  _CCCL_NODISCARD constexpr cudaMemPool_t get() const noexcept
+  {
+    return __pool_;
+  }
+};
+//! @brief Enables the \c device_accessible property for \c async_memory_resource.
+//! @relates async_memory_resource
+constexpr void get_property(async_memory_resource const&, _CUDA_VMR::device_accessible) noexcept {}
```

Comment:
i prefer this as a private friend function

Reply:
Me too, but this is the only way of having doxygen create documentation for it.

Reply:
i don't want to make the code worse for the sake of doxygen. i'd rather we conditionally compiled this depending on `DOXYGEN_SHOULD_IGNORE_THIS`. Doesn't have to be in this PR though.

Reply:
Done